### DISCUSSION TITLE
personalized turbine files to be changed with 3.1

### DISCUSSION THREAD

**Discussion by julienMs** [NatLabRockies/floris#415] [Q&A] [answered]
After updating to 3.1 I had some issue with the function turbine.get_power().
After a while I discovered that it was coming from that fact that I was using a personalized turbine file .yaml that worked for floris version 3.0 but not 3.1.
I had to adapt the file by adding the caracteristics of the turbine for the 0 and 50 m/s wind speed.

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi @julienMs,

In `v3.0`, we used to extrapolate the power production for situations that were not defined in the power-thrust tables. We realized that this could cause unpredictable behavior for high wind speeds. For example, if the user only specified the power and thrust coefficient curves between 4 and 15 m/s, then FLORIS `v3.0` would assume that the turbine produces the same amount of power at 50 m/s as it does at 15 m/s, and the same power at 0 m/s as it does at 4 m/s. There are plenty of situations imaginable where this is not correct.

Hence, rather than making an assumption on what the user intends for the power curve to do, we are enforcing users to specify the performance curve for all conditions they are evaluating in `v3.1`. Note that wake losses may create situations with very low wind speeds for waked turbines, and thus I am not surprised you were forced to specify the performance curves down to 0 m/s.

Best,

**Reply by rafmudaf** [reply]
FWIW This needs to be documented and I've added an issue to track these things: https://github.com/NREL/floris/issues/416
